011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/containers/PostgreSqlDatabaseTestResource.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.qe.containers;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.testcontainers.containers.GenericContainer;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+public class PostgreSqlDatabaseTestResource implements QuarkusTestResourceLifecycleManager {
+
+    public static final int PORT = 5432;
+
+    private static final String QUARKUS_DB_KIND = "quarkus.datasource.db-kind";
+    private static final String QUARKUS_DB_JDBC_URL = "quarkus.datasource.jdbc.url";
+    private static final String QUARKUS_DB_USER = "quarkus.datasource.username";
+    private static final String QUARKUS_DB_PASSWORD = "quarkus.datasource.password";
+    private static final String JDBC_URL_TEMPLATE = "jdbc:postgresql://%s:%s/%s?currentSchema=%s";
+    private static final String DEFAULT_DATABASE = "db";
+    private static final String DEFAULT_SCHEMA = "test";
+    private static final String DEFAULT_USER = "sarah";
+    private static final String DEFAULT_PASSWORD = "connor";
+    private static final String POSTGRESQL = "postgresql";
+
+    private GenericContainer<?> container;
+
+    @Override
+    public Map<String, String> start() {
```

Comment:
There is https://mvnrepository.com/artifact/org.testcontainers/postgresql/1.15.0

GenericContainer is not necessary to be used.

Reply:
Changed